src/modules/design/diagram-detail/diagram-detail.ts

```diff
@@ -350,6 +375,28 @@ export class DiagramDetail {
     }
   }
 
+  private async _openDirectory(): Promise<string> {
+    const canNotReadFromFileSystem: boolean = !this._canReadFromFileSystem();
+    if (canNotReadFromFileSystem) {
+      return;
+    }
+
+    this._ipcRenderer.send('save_single_diagram');
+
+    this._ipcRenderer.once('send_diagram_to_path', async(event: Event, openedFolder: File) => {
+      const noFolderSelected: boolean = openedFolder === null;
+      if (noFolderSelected) {
+        return;
+      }
+
+      await this.saveDiagram(openedFolder[0]);
+    });
+  }
+
+  private _canReadFromFileSystem(): boolean {
+    return (window as any).nodeRequire;
```

Comment:
Maybe we can to this check like so (looks less dirty imo):

```suggestion
    const isElectron: boolean = window.hasOwnProperty('nodeRequire');
    
    return isElectron;
```